lib/Doctrine/ORM/QueryBuilder.php

```diff
@@ -675,9 +678,9 @@ public function getMaxResults()
      * The available parts are: 'select', 'from', 'join', 'set', 'where',
      * 'groupBy', 'having' and 'orderBy'.
      *
-     * @param string    $dqlPartName
-     * @param Expr\Base $dqlPart
-     * @param bool      $append
+     * @param string       $dqlPartName The DQL part name.
+     * @param object|array $dqlPart     An Expr object.
```

Comment:
The types can not only be `Expr\Base`, but also `Expr\From`, `Expr\Comparison` and `Expr\OrderBy`, that do not extend `Expr\Base`. According to the source code, it can also be `array`.